File: seed/csharp-sdk/extra-properties/src/SeedExtraProperties/Types/Failure.cs
using System.Text.Json.Serialization;
using SeedExtraProperties.Core;

#nullable enable

namespace SeedExtraProperties;

public record Failure
{
    [JsonPropertyName("status")]
    public required string Status { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/extra-properties/src/Failure.php
<?php

namespace Seed;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class Failure extends JsonSerializableType
{
    /**
     * @var 'failure' $status
     */
    #[JsonProperty('status')]
    public string $status;

    /**
     * @param array{
     *   status: 'failure',
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->status = $values['status'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
